src/main/java/org/dependencytrack/tasks/scanners/SnykAnalysisTask.java

```diff
@@ -0,0 +1,467 @@
+                            if(problems!=null){
+                                vulnerability.setAliases(computeAliases(vulnerability, qm, problems));
+                            }
+                            final JSONArray cvssArray = vulnAttributes.optJSONArray("severities");
+                            vulnerability = cvssArray != null ? setCvssScore(cvssArray, vulnerability) : vulnerability;
+                            JSONArray coordinates = vulnAttributes.optJSONArray("coordinates");
+                            if (coordinates != null) {
+
+                                for (int countCoordinates = 0; countCoordinates < coordinates.length(); countCoordinates++) {
+                                    JSONArray representation = coordinates.getJSONObject(countCoordinates).optJSONArray("representation");
+                                    if (representation != null) {
+                                        vsList = parseVersionRanges(qm, purl, representation);
+                                    }
+                                }
+                            }
+                            qm.persist(vsList);
+                            Vulnerability synchronizedVulnerability = qm.synchronizeVulnerability(vulnerability, false);
+                            synchronizedVulnerability.setVulnerableSoftware(vsList);
+                            qm.persist(synchronizedVulnerability);
+                            addVulnerabilityToCache(component, synchronizedVulnerability);
+                            final Component componentPersisted = qm.getObjectByUuid(Component.class, component.getUuid());
+                            if (componentPersisted != null) {
+                                NotificationUtil.analyzeNotificationCriteria(qm, synchronizedVulnerability, componentPersisted, vulnerabilityAnalysisLevel);
+                                qm.addVulnerability(synchronizedVulnerability, componentPersisted, this.getAnalyzerIdentity());
+                                LOGGER.debug("Snyk vulnerability added : " + synchronizedVulnerability.getVulnId() + " to component " + component.getName());
+                            }
+                            Event.dispatch(new IndexEvent(IndexEvent.Action.COMMIT, Vulnerability.class));
+
+                        }
+                    }
+                }
+                updateAnalysisCacheStats(qm, Vulnerability.Source.SNYK, ApiBaseUrl, component.getPurl().toString(), component.getCacheResult());
+            } else if (responseCode == 404) {
+                Vulnerability vulnerability = new Vulnerability();
+                addVulnerabilityToCache(component, vulnerability);
+                updateAnalysisCacheStats(qm, Vulnerability.Source.SNYK, ApiBaseUrl, component.getPurl().toString(), component.getCacheResult());
+                handleUnexpectedHttpResponse(LOGGER, ApiBaseUrl, responseCode, statusText);
```

Comment:
The 404 response is already handled, there is no need to invoke the unexpected response handler anymore.